CryptoAnalysis/src/main/java/crypto/HeadlessCryptoScanner.java

```diff
@@ -302,9 +311,14 @@ protected CrySLAnalysisListener getAdditionalListener() {
 	protected List<CrySLRule> getRules() {
 		if (rules != null) {
 			return rules;
+		} else {
+			try {
+				return rules = CrySLRulesetSelector.makeFromRuleset("src/main/resources/JavaCryptographicArchitecture", RuleFormat.SOURCE, Ruleset.JavaCryptographicArchitecture);
+			} catch (CryptoAnalysisException e) {
+				LOGGER.error("Error happened when getting the CrySL rules from the specified directory: src/main/resources/JavaCryptographicArchitecture", e);
+			}
 		}
-		
-		return rules = CrySLRulesetSelector.makeFromRuleset("src/main/resources/JavaCryptographicArchitecture", RuleFormat.SOURCE, Ruleset.JavaCryptographicArchitecture);
+		return null;
```

Comment:
returning `null` here probably is dangerous because e.g. the consumer `SeedFactory` does not check for the null in its constructor. I'd prefer using `Collections.emptyList()` or throw an exceptions if you think this makes sense. I'd prefer returning empty.

Reply:
We will not throw here since this method is virtual and thus can be overwritten. throwing exceptions here changes behavior and also might break code.